src/services/ui/src/components/TimeoutModal/index.tsx

```diff
@@ -0,0 +1,87 @@
+import { useEffect, useState } from "react";
+import {
+  Button,
+  Dialog,
+  DialogContent,
+  DialogFooter,
+  DialogHeader,
+  DialogTitle,
+} from "@/components";
+import { Auth } from "aws-amplify";
+import { useIdle } from "@/hooks/useIdle";
+import { useGetUser } from "@/api";
+import { useCountdown } from "@/hooks/useCountdown";
+import { intervalToDuration } from "date-fns";
+
+const TWENTY_MINS_IN_MILS = 1000 * 60 * 20;
+const TEMP_5_SEC_IN_MILS = 5000;
+const TEN_MINS_IN_MILS = 60 * 10;
+
+export const TimeoutModal = () => {
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const isIdleForTwentyMins = useIdle(TEMP_5_SEC_IN_MILS, {
+    initialState: false,
+  });
+
+  const [count, { startCountdown, resetCountdown }] =
+    useCountdown(TEN_MINS_IN_MILS);
+  const { data: user } = useGetUser();
+
+  const onLogOut = () => {
+    Auth.signOut();
+  };
+
+  const onExtendSession = () => {
+    setIsModalOpen(false);
+
+    // artificial delay hiding the countdown reset after modal dismissal
+    setTimeout(() => {
+      resetCountdown();
+    }, 500);
+  };
+
+  useEffect(() => {
+    if (count === 0) {
+      onLogOut();
+    }
+  }, [count]);
+
+  useEffect(() => {
+    if (user?.user && isIdleForTwentyMins) {
+      startCountdown();
+      setIsModalOpen(true);
+    }
+  }, [isIdleForTwentyMins]);
+
+  const duration = intervalToDuration({ start: 0, end: count * 1000 });
+
+  return (
+    <Dialog open={isModalOpen} onOpenChange={onExtendSession}>
+      <DialogContent className="sm:max-w-[425px]">
+        <DialogHeader>
+          <DialogTitle>Session expiring soon</DialogTitle>
+        </DialogHeader>
+        <div className="py-4">
+          <span>
+            Due to inactivity, you&#39;ll be automatically logged out in{" "}
+            <strong>{duration.minutes}</strong> minutes and{" "}
+            <strong>{duration.seconds}</strong> seconds.
+          </span>
```

Comment:
We need to pluralize `minutes` and `seconds`. Do we have anything in our toolbox for this?

Reply:
Resolved with https://github.com/Enterprise-CMCS/macpro-mako/pull/610/commits/f19bb4f4cc799f6a38c6d390e6452e30fb261330